### PR TITLE
First implementation

### PR DESCRIPTION
**Pull request by pascal22p**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    scalaVersion := scala2_13,
+    organization := "uk.gov.hmrc",
+    crossScalaVersions := Seq(scala2_13),
+    libraryDependencies ++= BuildDependencies(),
+    isPublicArtefact := true,
+    majorVersion     := 0,
+    scalafmtOnCompile := true,
+    scalacOptions ++= Seq(
+      "-feature",
+      //"-Werror",
```
Comment (Christian-QA) [inline build.sbt]: This to be uncommented in a later PR?

File: src/main/uk/gov/hmrc/mongoFeatureToggles/internal/config/AppConfig.scala [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.mongoFeatureToggles.internal.config
+
+/*
```
Comment (Christian-QA) [inline src/main/uk/gov/hmrc/mongoFeatureToggles/internal/config/AppConfig.scala]: Double license

File: src/main/uk/gov/hmrc/mongoFeatureToggles/model/FeatureFlagNamesLibrary.scala [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import scala.collection.mutable.ListBuffer
+
+object FeatureFlagNamesLibrary extends FeatureFlagNamesLibrary
+
+trait FeatureFlagNamesLibrary {
+  private var features: ListBuffer[FeatureFlagName] = ListBuffer.empty[FeatureFlagName]
+
+  final def getAllFlags = {
+    if (features.isEmpty) {
+      throw new RuntimeException("No feature flags in FLagsLibrary. Have you added the flags?")
```
Comment (Christian-QA) [inline src/main/uk/gov/hmrc/mongoFeatureToggles/model/FeatureFlagNamesLibrary.scala]: 'FLagsLibrary' capital 'L'

File: build.sbt (hunk 1 above)
Comment (pascal22p) [reply build.sbt]: No. I forgot. Let me fix this.

File: src/main/uk/gov/hmrc/mongoFeatureToggles/internal/config/AppConfig.scala (hunk 2 above)
Comment (pascal22p) [reply src/main/uk/gov/hmrc/mongoFeatureToggles/internal/config/AppConfig.scala]: fixed

File: src/main/uk/gov/hmrc/mongoFeatureToggles/model/FeatureFlagNamesLibrary.scala (hunk 3 above)
Comment (pascal22p) [reply src/main/uk/gov/hmrc/mongoFeatureToggles/model/FeatureFlagNamesLibrary.scala]: fixed

**Review by Christian-QA — APPROVED** [review]

**Pull request merged**